lib/compute/jenkins-main-node.ts

```diff
@@ -366,17 +367,29 @@ export class JenkinsMainNode {
           + ' -i /hudson/authorizationStrategy/roleMap[2]/role -t attr -n "pattern" -v ".*"'
           + ' -s /hudson/authorizationStrategy/roleMap[2]/role -t elem -n permissions -v " "'
           // eslint-disable-next-line max-len
-          + `${JenkinsMainNodeConfig.rolePermissions().map((e) => ` -s /hudson/authorizationStrategy/roleMap[2]/role/permissions -t elem -n "permission" -v ${e}`).join(' ')}`
+          + `${JenkinsMainNodeConfig.adminRolePermissions().map((e) => ` -s /hudson/authorizationStrategy/roleMap[2]/role/permissions -t elem -n "permission" -v ${e}`).join(' ')}`
           + ' -s /hudson/authorizationStrategy/roleMap[2]/role -t elem -n "assignedSIDs" -v " " '
           // eslint-disable-next-line max-len
           + `${this.admins(oidcFederateProps.adminUsers).map(((e) => ` -s /hudson/authorizationStrategy/roleMap[2]/role/assignedSIDs -t elem -n "sid" -v ${e}`)).join(' ')}`
           + ' -s /hudson/authorizationStrategy --type elem -n roleMap'
           + ' -i /hudson/authorizationStrategy/roleMap[3] -t attr -n "type" -v "slaveRolesRoles"'
+          + ' -s /hudson/authorizationStrategy/roleMap[2] -t elem -n role -v " "'
+          + ' -i /hudson/authorizationStrategy/roleMap[2]/role[2] -t attr -n "name" -v "read-only"'
+          + ' -i /hudson/authorizationStrategy/roleMap[2]/role[2] -t attr -n "pattern" -v ".*"'
+          + ' -s /hudson/authorizationStrategy/roleMap[2]/role[2] -t elem -n permissions -v " "'
+          // eslint-disable-next-line max-len
+          + `${JenkinsMainNodeConfig.readOnlyRolePermissions().map((e) => ` -s /hudson/authorizationStrategy/roleMap[2]/role[2]/permissions -t elem -n "permission" -v ${e}`).join(' ')}`
+          + ' -s /hudson/authorizationStrategy/roleMap[2]/role[2] -t elem -n "assignedSIDs" -v " "'
+          + ' -s /hudson/authorizationStrategy/roleMap[2]/role[2]/assignedSIDs -t elem -n "sid" -v "anonymous"'
           + ' /var/lib/jenkins/config.xml'
```

Comment:
Given that the shell commands don't wait for previous commands to finish, this can potentially cause a problem if the config is being reloaded while the next shell command edits the file.

The best approach in my opinion here would be to try to reload the config only once after the file is edited (We'll need to test this manually first) and maybe even add sleep if required (given file editing is a slow operation).

Reply:
The config is reloaded in the next line. I tested this code and can see the changes as expected without any interruption

Reply:
Added sleep and tested. Working as expected.